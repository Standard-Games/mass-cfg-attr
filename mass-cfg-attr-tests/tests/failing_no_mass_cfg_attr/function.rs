#![deny(unused_must_use, deprecated)]
#![allow(path_statements)] // disable warning for test

#[deprecated]
#[must_use]
fn get_random_number() -> u8 {
    4 // chosen by fair dice roll
}

fn main() {
    // Violation of unused_must_use if not removed by mass_cfg_attr
    get_random_number();
}
