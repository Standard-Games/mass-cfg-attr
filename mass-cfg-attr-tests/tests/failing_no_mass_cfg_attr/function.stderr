error: use of deprecated function `get_random_number`
  --> tests/failing_no_mass_cfg_attr/function.rs:12:5
   |
12 |     get_random_number();
   |     ^^^^^^^^^^^^^^^^^
   |
note: the lint level is defined here
  --> tests/failing_no_mass_cfg_attr/function.rs:1:26
   |
1  | #![deny(unused_must_use, deprecated)]
   |                          ^^^^^^^^^^

error: unused return value of `get_random_number` that must be used
  --> tests/failing_no_mass_cfg_attr/function.rs:12:5
   |
12 |     get_random_number();
   |     ^^^^^^^^^^^^^^^^^^^^
   |
note: the lint level is defined here
  --> tests/failing_no_mass_cfg_attr/function.rs:1:9
   |
1  | #![deny(unused_must_use, deprecated)]
   |         ^^^^^^^^^^^^^^^
